crates/bitwarden-vault/src/cipher/cipher_risk_client.rs

```diff
@@ -0,0 +1,778 @@
+    /// ## HIBP Check Results (`exposed_result` field)
+    ///
+    /// The `exposed_result` field uses the `ExposedPasswordResult` enum with three possible states:
+    /// - `NotChecked`: Password exposure check was not performed because:
+    ///   - `check_exposed` option was `false`, or
+    ///   - Password was empty
+    /// - `Found(n)`: Successfully checked via HIBP API, password appears in `n` data breaches
+    /// - `Error(msg)`: HIBP API request failed with error message `msg`
+    ///
+    /// # Errors
+    ///
+    /// This method only returns `Err` for internal logic failures. HIBP API errors are
+    /// captured per-cipher in the `exposed_result` field as `ExposedPasswordResult::Error(msg)`.
+    pub async fn compute_risk(
+        &self,
+        login_details: Vec<CipherLoginDetails>,
+        options: CipherRiskOptions,
+    ) -> Result<Vec<CipherRisk>, CipherRiskError> {
+        // Wrap password_map in Arc to avoid cloning the HashMap for each future
+        let password_map = options.password_map.map(Arc::new);
+
+        // Create futures that can run concurrently
+        let futures = login_details.into_iter().map(|details| {
+            let http_client = self.client.internal.get_http_client().clone();
+            let password_map = password_map.clone();
+            let base_url = options
+                .hibp_base_url
+                .clone()
+                .unwrap_or_else(|| HIBP_DEFAULT_BASE_URL.to_string());
+
+            async move {
+                if details.password.is_empty() {
+                    // Skip empty passwords, return default risk values
+                    return CipherRisk {
+                        id: details.id,
+                        password_strength: 0,
+                        exposed_result: ExposedPasswordResult::NotChecked,
+                        reuse_count: None,
+                    };
+                }
+
+                let password_strength = Self::calculate_password_strength(
+                    &details.password,
+                    details.username.as_deref(),
+                );
+
+                // Check exposure via HIBP API if enabled
+                // Capture errors per-cipher instead of propagating them
+                let exposed_result = if options.check_exposed {
+                    match Self::check_password_exposed(&http_client, &details.password, &base_url)
+                        .await
+                    {
+                        Ok(count) => ExposedPasswordResult::Found(count),
+                        Err(e) => ExposedPasswordResult::Error(e.to_string()),
+                    }
+                } else {
+                    ExposedPasswordResult::NotChecked
+                };
+
+                // Check reuse from provided map
+                let reuse_count = if let Some(map) = &password_map {
+                    map.map.get(&details.password).copied()
+                } else {
+                    None
+                };
```

Comment:
:art: You could use option chaining methods here.

```suggestion
               let reuse_count = password_map.as_ref()
                                                 .and_then(|m| m.map.get(&details.password).copied());
```

Reply:
[1ea9850](https://github.com/bitwarden/sdk-internal/pull/499/commits/1ea98508fd1069a84e047325bab6f67e6c8c2ef5)